pkg/daemon/update.go

```diff
@@ -1726,6 +1731,57 @@ func (dn *Daemon) updateSSHKeys(newUsers []ign3types.PasswdUser) error {
 		}
 	}
 	if !dn.mock {
+		// On RHCOS8 and in older versions of OKD, the keys are written to `/home/core/.ssh/authorized_keys`.
+		// Newer versions of OKD will expect the keys at `/home/core/.ssh/authorized_keys.d/ignition`.
+		// Remove the keys file from the old path here. It will be recreated at the correct path by the
+		// atomicallyWriteSSHKey function called right after.
+		authKeyPath := filepath.Join(coreUserSSHPath, "authorized_keys")
+		authKeyFragmentDirPath := filepath.Join(coreUserSSHPath, "authorized_keys.d")
+
+		if dn.os.IsFCOS() {
+			// Check if the authorized_keys file exists. If it does, remove it.
+			_, err := os.Stat(authKeyPath)
+			if err == nil {
+				err := os.RemoveAll(authKeyPath)
```

Comment:
Since this is a file, I think just calling for Remove() should do.

Also, no strong opinion but I was wondering should we delete here entire file authorized_keys or delete ssh keys inside it that are managed by MCO? It could be possible that user have other keys added that without leveraging MCO.

Reply:
I was thinking about this, too, but I deemed it safer to use RemoveAll() just in case somebody had unexpectedly added a dir with that name here. Happy to change this though.

Regarding additional keys, is that supported today though? i.e. wouldn't those keys be overwritten by the MCD via `atomicallyWriteSSHKey` on update anyway?

Reply:
good point, MCO will overwrite the content during any node update.